spec/integration/util/windows/user_spec.rb

```diff
@@ -117,8 +123,21 @@ def expect_logon_failure_error(&block)
           Puppet::Util::Windows::User.logon_user(username, nil)
         }
       end
+
+      it 'should raise error given that logon returns false' do
+
+        allow(Puppet::Util::Windows::User).to receive(:logon_user_by_logon_type).with(
```

Comment:
I don't think that these checks are necessary. As a rule of thumb, only validate external behavior, not internal implementation details, like the calls to `logon_user_by_logon_type`

Reply:
I'm not sure you even need to add a new test here - this test does basically the same thing as an existing test.

I think the only thing that's would be worth testing is that `fLOGON32_LOGON_INTERACTIVE` is used when `fLOGON32_LOGON_NETWORK` fails. You have two ways to test this:

- Create a real account during the tests that doesn't support network, but does support interactive
- Intercept the call to `LogonUserW` and return `FFI::WIN32_FALSE` when supplied with `fLOGON32_LOGON_NETWORK`

If those seem too difficult, I think this code is probably fine to merge without any new tests given existing tests.

Reply:
@Iristyle I would let this test over just to check if both logons fail it will raise the expected error.
The rest of the tests just tests if the user has incorrect username or password.